source/MaterialXRuntime/Commands/PortCommands.cpp

```diff
@@ -0,0 +1,142 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Commands/PortCommands.h>
+#include <MaterialXRuntime/RtApi.h>
+
+#include <MaterialXRuntime/Private/PvtApi.h>
+#include <MaterialXRuntime/Private/PvtCommand.h>
+#include <MaterialXRuntime/Private/Commands/PvtSetPortValueCmd.h>
+#include <MaterialXRuntime/Private/Commands/PvtConnectionCmd.h>
+
+namespace MaterialX
+{
+namespace RtCommand
+{
+
+void setPortValueFromString(const RtPort& port, const string& valueString, RtCommandResult& result)
+{
+    // Use try/catch since the conversion from string might fail and throw.
+    try
+    {
+        RtValue v = RtValue::createNew(port.getType(), port.getParent());
+        RtValue::fromString(port.getType(), valueString, v);
+        PvtCommandPtr cmd = PvtSetPortValueCmd::create(port, v);
+        PvtApi::cast(RtApi::get())->getCommandEngine().execute(cmd, result);
+    }
+    catch (Exception& e)
+    {
+        result = RtCommandResult(false, e.what());
+    }
+}
+
+void setPortValue(const RtPort& port, bool value, RtCommandResult& result)
```

Comment:
If this was internal code I would agree. But since it's public API functions I think it's better to have them explicitly typed, to clearly present the range of value types that are available to set.